muduo/base/Condition.cc

```diff
@@ -8,12 +8,22 @@
 #include <errno.h>
 
 // returns true if time out, false otherwise.
-bool muduo::Condition::waitForSeconds(int seconds)
+bool muduo::Condition::waitForSeconds(double seconds)
 {
   struct timespec abstime;
   // FIXME: use CLOCK_MONOTONIC or CLOCK_MONOTONIC_RAW to prevent time rewind.
   clock_gettime(CLOCK_REALTIME, &abstime);
-  abstime.tv_sec += seconds;
+
+  const int64_t kNanoSecondsPerSecond = 1e9;
+  int64_t nanoseconds = static_cast<int64_t>(seconds * kNanoSecondsPerSecond);
+  if (1e5 > nanoseconds)
+  {
+    nanoseconds = 1e5;
+  }
+
+  abstime.tv_sec += static_cast<time_t>(nanoseconds / kNanoSecondsPerSecond);
+  abstime.tv_nsec = (abstime.tv_nsec + nanoseconds) % kNanoSecondsPerSecond;
```

Comment:
This is wrong.
abstime = 1234.5
seconds = 0.5
your result: abstime = 1234.1
correct result: 1235.1